### PR TITLE
Committed first working version of runoff. Includes brainstorming not…

### PR DESCRIPTION
**Pull request by The-Comet-Gaming**
…es, these will be removed upon review.

### CODE REVIEW

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+{
+    // Checking if the inputted name matches with any of the candidates
+    for (int loopCount = 0; loopCount < candidate_count; ++loopCount)
+    {
+        if (strcmp(candidates[loopCount].name, name) == 0)
+        {
+            preferences[voter][rank] = loopCount;
+            return true;
+        }
+    }
```
Comment (ChrisMellor) [inline src/c/problem set 3/lesson 2 - runoff/runoff.c:138]: Use the standard index named variable

**Review by ChrisMellor — COMMENTED** [review]
```c
if (strcmp(candidates[loopCount].name, name) == 0)
        {
            preferences[voter][rank] = loopCount;
            return true;
        }
```

This part looks weird, I'm not sure what you're trying do here, was this written by you or added by CS50?

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    if (voter_count > MAX_VOTERS)
+    {
+        printf("Maximum number of voters is %i\n", MAX_VOTERS);
+        return 3;
+    }
+
+    // Keep querying for votes
+    for (int i = 0; i < voter_count; i++)
+    {
+
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:68]: Yeah, that bit was written by CS50.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        // Query for each rank
+        for (int j = 0; j < candidate_count; j++)
+        {
+            string name = get_string("Rank %i: ", j + 1);
+
+            // Record vote, unless it's invalid
+            if (!vote(i, j, name))
+            {
+                printf("Invalid vote.\n");
+                return 4;
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:78]: This was written by CS50. I believe they are using `4` as an identifier to indicate where the program finished "incorrectly".

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    }
+
+    // Keep holding runoffs until winner exists
+    while (true)
+    {
+        // Calculate votes given remaining candidates
+        tabulate();
+
+        // Check if election has been won
+        bool won = print_winner();
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:92]: Yeah that one is CS50.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        if (won)
+        {
+            break;
+        }
+
+        // Eliminate last-place candidates
+        int min = find_min();
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:99]: Yeah this one is CS50 again.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        bool tie = is_tie(min);
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:100]: You know what I'm gonna say, CS50.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        // Eliminate anyone with minimum number of votes
+        eliminate(min);
+
+        // Reset vote counts back to zero
+        for (int i = 0; i < candidate_count; i++)
+        {
+            candidates[i].votes = 0;
+        }
+    }
+    return 0;
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:124]: Written by CS50. It's being used as an identifier for where the program exited it's run. `0` meaning a-okay basically.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c (hunk 1 above)
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:138]: Will do.

File: src/c/problem set 3/lesson 2 - runoff/runoff.c [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+            return true;
+        }
+    }
+
+    // loop that runs a number of times equal to the candidates
+    // if statement checking if any of the candidates.names matches the name input
+    // if a match is found, then set preferences[voter][rank] to equal loopCount.
+    // (The loop is to find a match by going through the candidates[loop].name)
+    // (if a match is found then the preferences[voter][rank] is assigned the loop value
+    // as it corresponds to the relevant candidate.)
```
[the remark this answers is not in the document]
Comment (The-Comet-Gaming) [reply src/c/problem set 3/lesson 2 - runoff/runoff.c:145]: I can start removing those once I've finished implementing a piece of code. Thank you.

**Review by The-Comet-Gaming — COMMENTED** [review]
I appreciate your feedback. I find it funny how most of what you had questions over weren't actually written by me. I hope that's a good thing and not just a case of my mistakes being overlooked over more glaring problems.

**Review by ChrisMellor — APPROVED** [review]

**Pull request merged**